src/runtime/storage.js

```diff
@@ -103,6 +105,78 @@ export class Storage {
       resolve();
     });
   }
+
+  /**
+   * Stores the current time to local storage, under the storageKey provided.
+   * Removes times older than a week in the process.
+   * @param {string} storageKey
+   */
+  storeEvent(storageKey) {
+    return this.get(storageKey, /* useLocalStorage */ true).then((value) => {
+      const dateValues = this.pruneDateArray_(
+        this.deserializeDateArray_(value)
+      );
+      dateValues.push(Date.now());
+      const valueToStore = this.serializeDateArray_(dateValues);
+      this.set(storageKey, valueToStore, /* useLocalStorage */ true);
+    });
+  }
+
+  /**
+   * Retrieves the current time to local storage, under the storageKey provided.
+   * Filters out timestamps older than a week.
+   * @param {string} storageKey
+   * @return {!Promise<!Array<number>>}
+   */
+  getEvent(storageKey) {
+    return this.get(storageKey, /* useLocalStorage */ true).then((value) =>
+      this.pruneDateArray_(this.deserializeDateArray_(value))
+    );
+  }
+
+  /**
+   * Converts a stored series of timestamps to an array of numbers.
+   * @param {?string} value
+   * @return {!Array<number>}
+   */
+  deserializeDateArray_(value) {
```

Comment:
Sorry, just realized this would be even clearer, since "date" within JavaScript would often refer to a `Date` instance, whereas "timestamp" usually refers to numbers
```suggestion
  deserializeTimestamps_(value) {
```